client/my-sites/post-relative-time-status/index.jsx

```diff
@@ -21,121 +20,184 @@ class PostRelativeTime extends React.PureComponent {
-				sameElse: this.props.translate( 'll [at] LT', {
-					comment:
-						'll refers to date (eg. 21 Apr) & LT refers to time (eg. 18:00) - "at" is translated',
-				} ),
+			extraStatusClassName = 'is-scheduled';
+			statusIcon = 'calendar';
+			const displayedTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'scheduled %(displayedTime)s', {
+				comment: '%(displayTime)s is when a scheduled post or page is set to be published',
+				args: {
+					displayedTime,
+				},
 			} );
-
-			const displayScheduleTime =
-				scheduledDate.diff( now, 'years' ) > 0
-					? scheduledTime
-					: scheduledTime.replace( scheduledDate.format( 'Y' ), '' );
-
-			statusText = this.props.translate( 'scheduled for %(displayScheduleTime)s', {
-				comment: '%(displayScheduleTime)s is when a scheduled post is set to be published',
+		} else if ( status === 'draft' || status === 'pending' ) {
+			const displayTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'draft last modified %(displayTime)s', {
+				comment: '%(displayTime)s is when a draft post or page was last modified',
 				args: {
-					displayScheduleTime,
+					displayTime,
 				},
 			} );
-			statusClassName += ' is-scheduled';
-			statusIcon = 'calendar';
-		} else if ( status === 'trash' ) {
-			statusText = this.props.translate( 'trashed' );
-			statusClassName += ' is-trash';
-			statusIcon = 'trash';
-		} else if ( this.props.includeBasicStatus ) {
-			if ( status === 'draft' ) {
-				statusText = this.props.translate( 'draft' );
-			} else if ( status === 'publish' ) {
-				statusText = this.props.translate( 'published' );
-			} else if ( status === 'new' ) {
-				statusText = this.props.translate( 'Publish immediately' );
-			}
+		} else if ( status === 'publish' ) {
+			const displayTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'published %(displayTime)s', {
+				comment: '%(displayTime)s is when a post or page was last modified',
+				args: {
+					displayTime,
+				},
+			} );
+		} else if ( status === 'private' ) {
+			const displayedTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'private last modified %(displayedTime)s', {
+				comment: '%(displayedTime)s is when a private post or page was last modified',
+				args: {
+					displayedTime,
+				},
+			} );
+		} else if ( status === 'new' ) {
+			statusText = this.props.translate( 'Publish immediately' );
 		}
 
+		return this.getLabel( statusText, extraStatusClassName, statusIcon );
+	}
+
+	/**
+	 * Get "sticky" label
+	 */
+	getStickyLabel() {
+		return this.getLabel( this.props.translate( 'sticky' ), 'is-sticky', 'bookmark-outline' );
+	}
+
+	/**
+	 * Get "Pending" label
+	 */
+	getPendingLabel() {
+		return this.getLabel( this.props.translate( 'pending review' ), 'is-pending' );
+	}
+
+	/**
+	 * Get Label for the status
+	 *
+	 * @param {*} statusText text status
```

Comment:
Missed these jsdoc param types on my previous review. Is there any way we can make them more specific? I think they all are `string`, aren't they?

Reply:
They weren't there last time :) I had to refactor part of the component...